src/SdkCommon/ClientRuntime/ClientRuntime/RestExceptionGeneric.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Runtime.Serialization;
+namespace Microsoft.Rest
+{
+#if FullNetFx
+    [Serializable]
+#endif
+    public class RestException<V> : System.Exception, IRestException
+    {
+        
+        /// <summary>
+        /// Initializes a new instance of the ErrorModelException class.
+        /// </summary>
+        public RestException()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the ErrorModelException class.
+        /// </summary>
+        /// <param name="message">The exception message.</param>
+        public RestException(string message)
+            : base(message, null)
+        {
+        }
+
+        public RestException(string message, System.Exception innerException)
+        : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the CloudException class caused by another exception.
+        /// </summary>
+        /// <param name="message">A description of the error.</param>
+        /// <param name="requestMessage">The request message.</param>
+        /// <param name="responseMessage">The response message.</param>
+        public RestException(string message, HttpRequestMessageWrapper requestMessage, HttpResponseMessageWrapper responseMessage, int statusCode) : base(message)
+        {
+            Request = requestMessage;
+            Response = responseMessage;
+            ResponseStatusCode = statusCode;
+        }
+
+        public V Body { get; private set; }
+        
+        /// <summary>
+        /// Gets information about the associated HTTP request.
+        /// </summary>
+        public HttpRequestMessageWrapper Request { get; set; }
+
+        /// <summary>
+        /// Gets information about the associated HTTP response.
+        /// </summary>
+        public HttpResponseMessageWrapper Response { get; set; }
+
+        /// <summary>
+        /// the status code returned by server
+        /// </summary>
+        public int ResponseStatusCode { get; set; }
+
+        public virtual void SetErrorModel(V model)
```

Comment:
@dsgouda once Body/ErrorModel has public setter, this function is no longer needed.